src/test/scala/org/broadinstitute/dsde/workbench/sam/util/NewRelicShadowResultReporterSpec.scala

```diff
@@ -0,0 +1,226 @@
+package org.broadinstitute.dsde.workbench.sam.util
+
+import akka.http.scaladsl.model.StatusCodes
+import cats.effect.IO
+import org.broadinstitute.dsde.workbench.model.{ErrorReport, ErrorReportSource, ValueObject, WorkbenchExceptionWithErrorReport}
+import org.broadinstitute.dsde.workbench.newrelic.NewRelicMetrics
+import org.scalatest.{FlatSpec, Matchers}
+import org.mockito.Mockito._
+import org.mockito.ArgumentMatchers._
+import org.scalatest.mockito.MockitoSugar
+import scala.concurrent.duration._
+
+class NewRelicShadowResultReporterSpec extends FlatSpec with Matchers with MockitoSugar {
+
+  private def createResultReporter = {
+    val newRelicMetrics = mock[NewRelicMetrics]
+    when(newRelicMetrics.gauge(any[String], any[Float])).thenReturn(IO.unit)
+    when(newRelicMetrics.incrementCounterIO(any[String], any[Int])).thenReturn(IO.unit)
+    new NewRelicShadowResultReporter("test", newRelicMetrics)
+  }
+
+  // match cases
+  "NewRelicShadowResultReporter" should "match complex case class structure" in {
+    val reporter = createResultReporter
+    val probe = TestCaseClass("asdfasdf", Seq(TestInnerCaseClass(MyValueObject("qqq"), 88), TestInnerCaseClass(MyValueObject("ppp"), 99)))
+    val result = reporter.resultsMatch(Right(probe), Right(probe))
```

Comment:
do one of these as `probe.copy()` to make sure you're not just testing memory location (and below also)